### PR TITLE
parsing of anisotropic B factors in MMCIFParser.py

### PR DESCRIPTION
**Pull request by svalqui**
This pull request addresses issue #1733, parsing of anisotropic B factors in Bio/PDB/MMCIFParser.py uses wrong mmCIF category

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by svalqui** [comment]
Hi BioPython Members,
Replaced _atom_site.aniso_U with _atom_site_anisotrop.U, and created an index by _atom_site_anisotrop.id, so it can be related back to _atom_site.id. to address issue #1733 .

Regards,

Sergio.

File: Bio/PDB/MMCIFParser.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def get_structure(self, structure_id, filename):
          - filename - name of mmCIF file, OR an open text mode file handle
 
         """
+
```
Comment (peterjc) [inline Bio/PDB/MMCIFParser.py]: You need to remove this new blank line, it is one of the reason the style checks on TravisCI are failing:

```
$ flake8 Bio/
Bio/PDB/MMCIFParser.py:52:1: D202 No blank lines allowed after function docstring
...
```

File: Bio/PDB/MMCIFParser.py [hunk 2]
```diff
@@ -99,16 +101,25 @@ def _build_structure(self, structure_id):
...
         except KeyError:
             # no anisotropic B factors
             aniso_flag = 0
+
+        if aniso_flag == 1:
+            # create aniso_dict indexed by _atom_site_anisotrop.id
+            aniso_dict = {}
+            for idx, val in enumerate(mmcif_dict["_atom_site_anisotrop.id"]):
+                aniso_dict[val.strip()] = (aniso_u11[idx], aniso_u12[idx],
+                                   aniso_u13[idx], aniso_u22[idx], aniso_u23[idx],
```
Comment (peterjc) [inline Bio/PDB/MMCIFParser.py]: You need to fix the indentation here,

```
$flake8 Bio/
...
Bio/PDB/MMCIFParser.py:120:36: E128 continuation line under-indented for visual indent
Bio/PDB/MMCIFParser.py:121:36: E128 continuation line under-indented for visual indent
...
```

File: Bio/PDB/MMCIFParser.py [hunk 3]
```diff
@@ -337,17 +348,25 @@ def _build_structure(self, structure_id, filehandle):
...
         except KeyError:
             # no anisotropic B factors
             aniso_flag = 0
 
+        if aniso_flag == 1:
+            # create aniso_dict indexed by _atom_site_anisotrop.id
+            aniso_dict = {}
+            for idx, val in enumerate(mmcif_dict["_atom_site_anisotrop.id"]):
+                aniso_dict[val.strip()] = (aniso_u11[idx], aniso_u12[idx],
+                                   aniso_u13[idx], aniso_u22[idx], aniso_u23[idx],
```
Comment (peterjc) [inline Bio/PDB/MMCIFParser.py]: Again, indentation needs fixing:

```
$ flake8 Bio/
...
Bio/PDB/MMCIFParser.py:367:36: E128 continuation line under-indented for visual indent
Bio/PDB/MMCIFParser.py:368:36: E128 continuation line under-indented for visual indent
```

**Comment by svalqui** [comment]
@peterjc Hi Peter, Flake8 issues corrected, Thank You.

**Comment by peterjc** [comment]
@jgreener64 could you review this please?

**Comment by jgreener64** [comment]
This is similar to https://github.com/biopython/biopython/pull/1775, which looks like it didn't finish review. I don't know what the protocol is for two similar PRs but that one came first and has tests.

Either way, they are quite similar. This one needs tests though.

**Comment by peterjc** [comment]
Thanks Joe. Sorry @svalqui &  @rtf-const, I didn't spot the duplication of effort here, but would have wanted to see tests either way. Given #1775 is older and already has tests, focusing on that seems sensible to resolve issue #1733.

@svalqui does #1775 fix your use case? Is there anything in your work in #1822 you think is worth adding to #1775?

**Comment by peterjc** [comment]
I have now merged #1775, but would like to know if there is anything worth incorporating from this branch as well - or should we close it @svalqui? Thank you!

**Comment by svalqui** [comment]
Hi All,

@rtf-const  I saw your comments in the trail before but I missed reading it carefully and thought of it as a typo only on the test file; my apologies for not been more thorough and notice the complete work.

I agree 1775 is a better piece of work and was logged first so is good that is merged now.

I tested this with another cif, and has given me unexpected results, @peterjc You can disregard this PR.

All good.

**Comment by peterjc** [comment]
Did you mean to say your tests had "unexpected results", or "expected results"? If there is still a problem here please do open a new issue on this, but for now I will close this pull request. Thank you for your input though, and checking the alternative code.

**Pull request closed without merging**